### PR TITLE
Feature/add requestgroups pages

### PR DESCRIPTION
**Pull request by eheinrich**
Pulled over components for the compose, request detail, and requestgroup detail pages.

I added a navigation guard in the router file that redirects a user to the login page if a route has been marked as requiring authentication but the user is not logged in. After logging in, the user will be redirected to the page they originally tried to load. So far for the pages that have been pulled over, this is needed for the compose page.

Added archive token authentication for loading thumbnails/ archive data.

This branch is deployed on http://observation-portal-separate-frontend.lco.gtn/

### CODE REVIEW

File: src/components/RequestgroupHeader.vue [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+          type: 'POST',
+          url: this.observationPortalApiUrl + '/api/requestgroups/' + this.requestgroup.id + '/cancel/',
+          contentType: 'application/json',
+          success: function() {
+            // TODO: Redirect with router
+            window.location = '/requestgroups/' + that.requestgroup.id + '/';
+          },
+          error: function(response) {
+            if(response.status === 429) {
+              alert('Too many cancel requests, your request to cancel has been throttled. Please contact support.');
```
Comment (jnation3406) [inline src/components/RequestgroupHeader.vue:138]: Maybe say "Your account has submitted too many cancel requests in a day, so your request to cancel has been throttled. Please contact support." unless that doesn't fit well?

File: src/main.js [hunk 2]
```diff
@@ -30,7 +23,29 @@ getRuntimeConfig().then(function(json) {
...
+        xhr.setRequestHeader('X-CSRFToken', csrftoken);
+      }
+      settings.xhrFields = {
+        withCredentials: true
+      };
+    }
+  });
+
+  // Add the archive token to a request being sent to the archive api or the thumbservice
+  $.ajaxPrefilter(function (options, originalOptions, jqXHR) {
```
Comment (jnation3406) [inline src/main.js:45]: this is pretty cool

File: src/components/Drafts.vue [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        return this.items.length < 1;
+      },
+      rows: function() {
+        return this.items.length;
+      }
+    },
+    methods: {
+      fetchDrafts: function() {
+        let that = this;
+        $.getJSON(this.observationPortalApiUrl + '/api/drafts/?limit=100', function(data) {
```
[the remark this answers is not in the document]
Comment (eheinrich) [reply src/components/Drafts.vue:104]: The pagination actually happens in memory... there's only one request sent off to get drafts, and all those drafts returned are then paginated. I'm guessing that's why the limit was put in?

File: src/components/RequestgroupHeader.vue (hunk 1 above)
Comment (eheinrich) [reply src/components/RequestgroupHeader.vue:138]: Will change

File: src/archive.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import $ from 'jquery';
+import 'jquery-file-download';
+
+export { downloadZip, downloadAll, getLatestFrame };
+
+function downloadZip(frameIds, archiveRoot, archiveToken) {
+  let postData = {};
+  for (let i = 0; i < frameIds.length; i++) {
+    postData['frame_ids[' + i + ']'] = frameIds[i];
```
[the remark this answers is not in the document]
Comment (eheinrich) [reply src/archive.js:9]: Hmm, the form data that is sent along with the POST is something like:
```
frame_ids[0]: 1234567
frame_ids[1]: 1234568
frame_ids[2]: 1234569
frame_ids[3]: 1234560
frame_ids[4]: 1234561
frame_ids[5]: 1234562
```
So it's definitely not an array, and the endpoint must be ok with this... I'm not sure why it was done this way originally, I'll look into it a bit.

[the remark this answers is not in the document]
Comment (eheinrich) [reply src/archive.js:9]: I guess this is a workaround for [lists being disabled as an HTML input](https://github.com/encode/django-rest-framework/pull/5702) in DRF - I did indeed just try your suggestion, and got a 400, and a message in the response of "Lists are not currently supported in HTML".

And the reason that this download is initiated via an HTML form is described in detail in the [blog post about the jquery fileDownload plugin](http://johnculviner.com/jquery-file-download-plugin-for-ajax-like-feature-rich-file-downloads/) TL;DR the plugin uses a hidden iframe to initiate the download, and makes [hidden inputs](https://github.com/johnculviner/jquery.fileDownload/blob/master/src/Scripts/jquery.fileDownload.js#L313) for each item in the data that is sent along with the request.

Submitting the frame ids as is being done, they end up get parsed out as a list: https://github.com/encode/django-rest-framework/blob/master/rest_framework/utils/html.py#L15

I can add a comment to the code.

[the remark this answers is not in the document]
Comment (jnation3406) [reply src/archive.js:13]: oh javascript

File: src/components/Drafts.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (jnation3406) [reply src/components/Drafts.vue:104]: yeah, and honestly 100 is plenty since they can delete. Who would ever need more than that to paginate through

**Pull request merged**